public/services/data_source_service.ts

```diff
@@ -74,24 +74,23 @@ export class DataSourceService {
   }
 
   setDataSourceId(newDataSourceId: string | null) {
-    if (this.dataSourceId$.getValue() === newDataSourceId) {
-      return;
-    }
```

Comment:
Could you elaborate more on why we remove duplicate check in set function?

Reply:
Since we already add `distinctUntilChanged` in `getDataSourceId$`, the next value won't be the same. Then we can remove the duplicate check here.